docs/DeveloperGuide.adoc

```diff
@@ -222,42 +226,216 @@ image::CommitActivityDiagram.png[]
+`Arms Day`. The upcoming set of the next exercise is displayed for the user.
+
+Step 2. The user completes the first set and types `done`.
+
+* `DoneCommand#execute()` is invoked, which in turn calls `OngoingSession#done()` to obtain the next set and statefully
+updates the instance of `OngoingSession`.
+
+* The rest timer starts counting from 00:00.
+
+.DoneCommand sequence diagram
+image::DoneCommandSequenceDiagram.png[]
+
+Step 3. The user fails the last and final set of `Benchpress` and types `skip`.
+
+* `SkipCommand#execute()` is invoked. The instance of `OngoingSession` is updated. Since `OngoingSession#hasSetLeft()`
+returns false, `OngoingSession#finish()` is called which creates and returns an immutable `Session` object. This object
+is saved into the `sessionList` of `ModelManager` for saving and use by the `Log` feature.
+
+[Note] In the usual flow, the `stop` command is not used, and is reserved only for a premature (incomplete) ending of a
+session.
+
+==== Design Considerations
+===== Aspect: Seamless user-first experience
+* **Current Implementation:** Minimal commands and typing during an ongoing session.
+** Pros: Less interruption during actual exercising to enter commands.
+** Cons: More experienced users are not able to have a more customizable workout.
+
+
+===== Aspect: Beginner-friendly automatic set-ordering
+* **Current Implementation:** Exercises and sets have fixed ordering based on their creation.
+** Pros: Easier for new user to follow a session, less typing required during ongoing session.
+** Cons: Experienced users are not able to have an ad-hoc customizable workout without creating a new workout with the specific order they want.
+// end::session[]
+
+// tag::exercise[]
+=== Exercise
+==== Implementation
+The exercise feature forms the basic building block for the application. It allows users to CRUD exercises, which will then be used for the creation of workouts. The command is prefixed by the keyword `exercise`. Available commands are `create`, `edit`, `list`, `delete` and many more.
+
+.Exercise's CreateCommand Sequence Diagram
+image::exercise/CreateCommandSequenceDiagram.png[]
+
+In this portion, we will trace the sequence diagram of the `exercise create` command to better understand the internals of the Exercise feature.
+
+. The user enters the command `exercise create e/Bench Press`
+. LogicManager will pass the command to the ParserManager for parsing
+. ParserManager upon seeing that the command is prefixed by `exercise` creates a ExerciseCommandParser
+. ParserManager then pass `create e/Bench Press` to ExerciseCommandParser
+. ExerciseCommandParser upon seeing that the command is prefixed by `create` creates a CreateCommandParser
+. ExerciseCommandParser then pass the argument `e/Bench Press` to CreateCommandParser
+. CreateCommandParser then attempts to create an ExerciseName object using the String in the argument
```

Comment:
Since you have multiple `CreateCommand` classes, one for each package, it might be better to use the package name along with the class name here.